src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -308,11 +308,29 @@ private ProcessPriorityClass PriorityClassCore
         }
 
         /// <summary>Checks whether the argument is a direct child of this process.</summary>
-        private bool IsParentOf(Process possibleChildProcess) =>
-            Id == possibleChildProcess.ParentProcessId;
+        private bool IsParentOf(Process possibleChildProcess)
+        {
+            try
+            {
+                return Id == possibleChildProcess.ParentProcessId;
+            }
+            catch (Exception e) when (IsProcessInvalidException(e))
```

Comment:
Could the code also be changed to not use exception filters

Reply:
I agree with @jkotas that going through and removing exception filters is a step in the wrong direction.
https://github.com/dotnet/runtime/issues/50281#issuecomment-808317762

Reply:
What is the benefit of using generic Exception with filters over catch for specific exceptions?

Reply:
All of the places that handle these exceptions then need to duplicate the list.  And each of them needs its own catch block with the logic duplicated.

Reply:
I think that only hides the real problem here. The better improvement would be not to use exceptions for normal flow execution here. For example by changing ParentProcessId to return int? and catch the only exceptions which can really happen on specific platforms instead of generic IsProcessInvalidException.

Reply:
> The better improvement would be not to use exceptions for normal flow execution here.

It's not using exceptions for "normal flow" here.  It's using exceptions for exceptional situations, e.g. a race condition between a process being alive one moment and then not the next.  Making a blanket statement that exceptions shouldn't be used for control flow is akin to saying code should never have a try/catch and all throws should instead be fail-fasts.  That ship sailed over two decades ago.

It's also entirely orthogonal to this PR.  This PR is not introducing any new exceptions in any new places, nor is it eating exceptions any more or any less than before.  It's only changing the manner in which those same exceptions are caught.

Reply:
> It's also entirely orthogonal to this PR. 

Fair enough. I didn't want to block this just pointing out possible better implementation. I might give it a shot once this lands.